### PR TITLE
update advice/examples and requirements to reflect document outline reality

### PR DESCRIPTION
**Pull request by stevefaulkner**
refer to  https://github.com/w3c/html/issues/33

This PR includes:
- additions to sectioning elements about how semantics are conveyed to users by AT
- cleanup of examples using headings to reflect reality of how they are conveyed (by heading elevel h1-h6) not sectioning nesting level.
- changed to SHOULD for authors to use heading of appropriate level
- minor code clean up

### CODE REVIEW

File: sections/semantics-sections.include [hunk 1]
```diff
@@ -293,10 +293,20 @@
...
+  can be announced by screen reader software when a user navigates to an <{section}> element. User Agents 
+  may also provide methods to navigate to <{section}> elements.</p>
 
   <div class="example">
     In the following example, we see an article (part of a larger Web page) about apples,
-    containing two short sections.
+    containing two short sections. 
+	
+	<p class="note">As the <{section}> has an <code>aria-label</code> attribute providing a brief description of 
+	the contents. Assistive technology may convey the <a value for="role"><code>region</code></a> role
```
Comment (chaals) [inline sections/semantics-sections.include:308]: Grammar nit. Either 

> …contents, assistive…

and use "can" not "may",

Or remove "As" from the beginning of the sentence.

File: sections/semantics-sections.include [hunk 2]
```diff
@@ -449,6 +460,12 @@
 
   The <{nav}> element <a>represents</a> a section of a page that links to other pages or to
   parts within the page: a section with navigation links.
+  
+   <p>The semantics of the <{nav}> element may be conveyed to users of Assistive Technology.
```
Comment (chaals) [inline sections/semantics-sections.include]: We prefer active voice:

> Assistive Technology may convey the semantics of the <{nav}> element to users.

…

Not critical for now, but in general we should try to get this consistent.

File: sections/semantics-sections.include [hunk 3]
```diff
@@ -820,10 +843,16 @@
...
-  <code>h1</code>–<code>h6</code> elements must not be used to markup subheadings, subtitles,
+  <code>h2</code>–<code>h6</code> elements must not be used to markup subheadings, subtitles,
   alternative titles and taglines unless intended to be the heading for a new section or subsection.
   Instead use the markup patterns in the [[#common-idioms-without-dedicated-elements]] section of
   the specification.
+  
+  <p>The semantics of the <code>h1</code>–<code>h6</code> elements may be conveyed to users of Assistive Technology. 
+  This information can provide a hint to users as to the type of content. For example the <a>role</a> of
+   the element, which in this case is "heading" and the heading level "1" to "6", can be 
+   announced by screen reader software when a user navigates to an <code>h1</code>–<code>h6</code> element.
```
Comment (chaals) [inline sections/semantics-sections.include:854]: I would rewrite this:

"Assistive technology often announces the presence and level of a heading to users, as a hint to understand the structure of a document and construct a 'mental model' of its outline."

File: sections/semantics-sections.include [hunk 4]
```diff
@@ -1341,7 +1384,7 @@
     ("Grunt") is back at the top level.
   </div>
 
-  Sections may contain headings of any <a>rank</a>, and authors are strongly encouraged to use
+  Sections may contain headings of a <a>rank</a> equal to their section nesting level, authors should use
```
Comment (chaals) [inline sections/semantics-sections.include]: split the sentence where you currently have a comma.

File: sections/semantics-sections.include [hunk 5]
```diff
@@ -1397,7 +1440,7 @@
     There are currently no known implementations of the outline algorithm in graphical browsers or
```
Comment (chaals) [inline sections/semantics-sections.include]: Hmm. There are extensions that do it, there are just no native outline implementations I am aware of. It's an important difference.

**Comment by chaals** [comment]
I think you've got something weird in your setup, and it isn't rebasing properly so we continue to get all your old commits included.

I've listed a handful of comments, but the only one I think _must_ be addressed is regarding the implementations - your actual changes all look good to me.

File: sections/semantics-sections.include [hunk 6]
```diff
@@ -820,10 +843,16 @@
   element is said to have the highest rank, the <code>h6</code> element has the lowest rank, and two
   elements with the same name have equal rank.
 
-  <code>h1</code>–<code>h6</code> elements must not be used to markup subheadings, subtitles,
+  <code>h2</code>–<code>h6</code> elements must not be used to markup subheadings, subtitles,
```
Comment (travisleithead) [inline sections/semantics-sections.include:846]: Why is this an h2 -- bug or by design?

Comment (stevefaulkner) [inline sections/semantics-sections.include:846]: a subheading is subordinate to a heading, so should not be a `<h1>`

**Comment by ChristianGrete** [comment]
Even if this matches the reality somehow better, I really don’t understand this approach as these changes degrade sectioning elements and especially the `<section>` element to grouping containers that only indicate the scope of a section rather than defining a new one. This makes it almost impossible to build websites out of flexible and reusable dynamic modules without being limited to 6 heading levels and also eliminates the freedom to render modules at any desired place.

I was thinking that this spec describes what HTML renderers should implement in future to be compliant and not just reflects what is currently possible.

**Comment by stevefaulkner** [comment]
On 7 July 2016 at 15:05, Christian Grete notifications@github.com wrote:

> I was thinking that this spec describes what HTML renderers should
> implement in future to be compliant and not just reflects what is currently
> possible.

The spec did do that for 5+ years, but implementers didn't implement and
the authoring advice based on what could.but didn't happen, was having a
negative impact on the real world semantics being conveyed to users.

## 

Regards

SteveF
Current Standards Work @W3C
http://www.paciellogroup.com/blog/2015/03/current-standards-work-at-w3c/

**Comment by ChristianGrete** [comment]
I understand the _why_ behind that move but then tell me how to embed stand-alone content (e.g. a whole book or a magazine article) into an HTML document without changing its original sectioning tree? Should one switch over to `<iframe>` or web components just for this purpose?

I think the better approach would be to recommend what is currently possible rather than giving up the original idea of the `<section>` element and deleting all the examples.

**Comment by adanilo** [comment]
The spec. when reaching Recommendation state should reflect what _is_ implemented by at least 2 implementations. So if something isn't implemented, it shouldn't be in the spec. So "what HTML renderers should implement in future" is not the purpose of the spec. For things that should be implemented in future, the proper forum for that is to bring the feature into the WICG for incubation and if it gains traction, as in implementations then it'll make it into a future version of the spec.

**Comment by ChristianGrete** [comment]
@adanilo I see. Thank you for your explanation.

**Comment by chaals** [comment]
@ChristianGrete to answer the question of how to embed stuff, unless browsers decide to implement something more useful, the most effective approach is probably to write a short javascript that identifies the heading level where you are inserting the content, and then changes the headings within the content.

You can trigger this with e.g. a data-inserted attribute. An obvious limitation is that you can only get to six levels total, which is problematic for "professional" use but probably good enough for most simple use case.

**Comment by ChristianGrete** [comment]
@chaals So, the answer is that it is currently impossible and that one has to actually change the original sectioning tree. I’d rather recommend using an `<iframe>` in this case.

**Pull request merged**